### PR TITLE
Issue #1: Finished basic templates flow and project config (templates are in progress)

### LINKED ISSUE #1: Setup templates environment and assets workflow

### PR DESCRIPTION
**Pull request by lilliputten**
This PR is 'technical' and doesn't required review.

But here is two controversal things:

- I had to move templates into the `dds_registration/templates` folder due to problems on my demo stand (it was impossible to set django to get templates from the `dds_registration/templates/dds_registration` folder, for some reasons).

- When I tried to install all the dependencies on my demo stand (Gentoo Base System release 2.7 linux, with `pip install .`) the copy of the project had installed into the venv folder and when I did `from dds_registration.settings import *` to configure wsgi startup entry all the sources has taken from that location, not from my sources. I couldn't fix it and started to use `pip install` in the manual mode, with finishing `pip uninstall -y dds_registration`. UP: I see that the same thing happens in the local environment (but here it doesn't cause those problems.). What is the correct way to install required dependencies?

All finished tasks:

- Updated project environment (customized settings, added misc config files), added core management utitlities (clearcache), context processors, basic templates.
- Added static assets (webfonts, bootstrap, django ejected resources).
- Added bootstrap scss source files.
- Added scss preprocessing (`django-libsass`), .gitignore exclusion for cache files, example block with a template and styles (`static/blocks/test`), added webfonts and bootstrap resources.
- Added wsgi configuration for demo production host.
- Updated production configuration.
- Moved templates to the root of templates folder (due to `TemplateDoesNotExist ` error on the prod server).
- Added project build variables (verion, timestamps).

### CODE REVIEW

**Comment by cmutel** [comment]
> I had to move templates into the dds_registration/templates folder due to problems on my demo stand (it was impossible to set django to get templates from the dds_registration/templates/dds_registration folder, for some reasons).

In general the PR doesn't have the right folder structure, but that is fine. I am used to fixing path problems, I will figure it out 😄. It's fine for now.

 > When I tried to install all the dependencies on my demo stand (Gentoo Base System release 2.7 linux, with pip install .) the copy of the project had installed into the venv folder and when I did from dds_registration.settings import * to configure wsgi startup entry all the sources has taken from that location, not from my sources. I couldn't fix it and started to use pip install in the manual mode, with finishing pip uninstall -y dds_registration. UP: I see that the same thing happens in the local environment (but here it doesn't cause those problems.). What is the correct way to install required dependencies?
 
 Running `pip install .` will *copy* the python library over to your `site-packages` directory, but only the assets defined in either `pyproject.toml` or `MANIFEST.in`. See above. You can force the CWD to be installed with `pip install -e .`.

**Comment by cmutel** [comment]
> Added scss preprocessing (django-libsass), .gitignore exclusion for cache files, example block with a template and styles (static/blocks/test), added webfonts and bootstrap resources.

- [x] @lilliputten Please add some quick documentation on what (if anything) is needed to get the preprocessor to run before starting the webapp.

File: index.wsgi [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+with open(activate_this) as f:
+    code = compile(f.read(), activate_this, 'exec')
+    exec(code, dict(__file__=activate_this))
+
+# Inject application path...
+sys.path.insert(1, rootPath)
+
+import django
+
+if django.VERSION[1] <= 6 and django.VERSION[0] <= 1:
```
Comment (cmutel) [inline index.wsgi]: We are Django 5+ - why do we need code here for 1.6?

**Comment by lilliputten** [comment]
> > Added scss preprocessing (django-libsass), .gitignore exclusion for cache files, example block with a template and styles (static/blocks/test), added webfonts and bootstrap resources.
> 
>     * [ ]  @lilliputten Please add some quick documentation on what (if anything) is needed to get the preprocessor to run before starting the webapp.

It isn't required: all those things work inside the django, with `django-libsass`, and it's already added in the dependencies list. And I'm go to use the same scheme with scripts (if we'll need them; highly likely that, we'll do).

**Comment by cmutel** [comment]
> Added wsgi configuration for demo production host.

- [x] The `index.wsgi` file you added doesn't belong to this repo - it duplicates the generic [wsgi.py](https://github.com/Depart-de-Sentier/dds_registration/blob/main/dds_registration/wsgi.py) we already have, and doesn't some path stuff that shouldn't be necessary once we get the packaging sorted. Fine to have it for your local deploy though.

File: index.wsgi (hunk 1 above)
Comment (lilliputten) [reply index.wsgi]: > We are Django 5+ - why do we need code here for 1.6?

Ok, just removed it. But I'm not sure that this file is required at all: I use it only to test the app on my demo server.

And, if we'll use it, then we'll need at least to add configuration to allow choosing the proper venv name/location or to don't use it at all.

**Comment by lilliputten** [comment]
> > Added scss preprocessing (django-libsass), .gitignore exclusion for cache files, example block with a template and styles (static/blocks/test), added webfonts and bootstrap resources.
> 
> * [x]  @lilliputten Please add some quick documentation on what (if anything) is needed to get the preprocessor to run before starting the webapp.

Probably, I was wrong: that libsass (https://github.com/sass/libsass) used in `django_libsass.SassCompiler` (https://github.com/torchbox/django-libsass) is deprecated (https://sass-lang.com/blog/libsass-is-deprecated/).

Going to rewrite the flow setup with sass filter in `COMPRESS_PRECOMPILERS` -- it'll require installed sass compiler (probably, with node). I'll add a readme when a solution is found.

UP: Tell me pls, is it possible to install sass binary from `pyproject.toml` (there are some: https://github.com/sass/dart-sass/releases/latest) or to invoke npm installation (to run `npm install`, for example)?

**Comment by cmutel** [comment]
You are making my life difficult 😛. It [doesn't seem like there is a great option](https://www.reddit.com/r/flask/comments/1ak8xrh/dartsass_python_wrapper/).

So, we have a couple of options:
1. Even though libsass is deprecated, is [is still being maintained](https://github.com/sass/libsass), and the [python library is as well](https://pypi.org/project/libsass/).
2. There is a python wrapper for [dart sass](https://github.com/dumdoo/py-dart-sass), but that seems dead.
3. Compilation of the assets can be done independently and then checked into the repo.

**Comment by lilliputten** [comment]
> You are making my life difficult 😛. It [doesn't seem like there is a great option](https://www.reddit.com/r/flask/comments/1ak8xrh/dartsass_python_wrapper/).
> 
> So, we have a couple of options:
> 
> 1. Even though libsass is deprecated, is [is still being maintained](https://github.com/sass/libsass), and the [python library is as well](https://pypi.org/project/libsass/).
> 2. There is a python wrapper for [dart sass](https://github.com/dumdoo/py-dart-sass), but that seems dead.
> 3. Compilation of the assets can be done independently and then checked into the repo.

Here is the solution for the option 3 above -- a manual compilation of the assets (with a small readme):

https://github.com/Depart-de-Sentier/dds_registration/pull/4/commits/7ef2de76e10d71d46a9deb858589c110a7316858

**Pull request merged**